### PR TITLE
Clickable Row

### PR DESCRIPTION
**Pull request by rsumilang**
The PR is an example of the assets search screen as having each row clickable.

- Added NodePolyfillPlugin because MimeTypes required a path.extname polyfill

### CODE REVIEW

File: src/plugins/dam/components/search-assets-screen/index.js [hunk 1]
```diff
@@ -98,10 +100,10 @@ function SearchAssetsScreen(props) {
                 const transcodingStatus = node.get('transcoding_status');
 
                 return (
-                  <tr key={`${node.get('_id')}`} className={`status-${node.get('status')}`}>
+                  <tr key={`${node.get('_id')}`} className={`status-${node.get('status')}`} style={{cursor: 'pointer'}}>
```
Comment (gdbrown) [inline src/plugins/dam/components/search-assets-screen/index.js:103]: @rsumilang use the `cursor-pointer` utility class in className instead.

File: src/plugins/dam/components/search-assets-screen/index.js [hunk 2]
```diff
@@ -98,10 +100,10 @@ function SearchAssetsScreen(props) {
...
                     <td><Input type="checkbox" onChange={() => batch.toggle(node)} checked={batch.has(node)} /></td>
                     <td className="text-center"><AssetIcon id={node.get('_id')} /></td>
-                    <td>
+                    <td onClick={() => navigate(nodeUrl(node, 'view'))}>
```
Comment (gdbrown) [inline src/plugins/dam/components/search-assets-screen/index.js:106]: @rsumilang can we make an onClick function and put it on the row (one time).  accept the event and inspect the target to see if it has a data-ignore-row-click.  then put the data-ignore-row-click on the td we don't need a handler for every td?

**Review by gdbrown — CHANGES_REQUESTED** [review]

File: src/plugins/dam/components/search-assets-screen/index.js (hunk 1 above)
Comment (rsumilang) [reply src/plugins/dam/components/search-assets-screen/index.js]: @gdbrown there seems to be a bug in chrome when using the class name instead. the `cursor: pointer` setting is inherited down to each `td` but the cursor does not change. only changes when using the class if applied to each individual `td`.

File: src/plugins/dam/components/search-assets-screen/index.js (hunk 2 above)
Comment (rsumilang) [reply src/plugins/dam/components/search-assets-screen/index.js]: ok check it out

File: src/plugins/dam/components/search-assets-screen/index.js [hunk 3]
```diff
@@ -15,6 +16,7 @@ import SearchForm from '@triniti/cms/plugins/dam/components/search-assets-screen
 import BatchOperationsCard from '@triniti/cms/plugins/dam/components/search-assets-screen/BatchOperationsCard.js';
 import useBatch from '@triniti/cms/plugins/ncr/components/useBatch.js';
 import AssetIcon from '@triniti/cms/plugins/dam/components/asset-icon/index.js';
+import nodeClickNavigator from '@triniti/cms/utils/nodeClickNavigator.js';
```
Comment (gdbrown) [inline src/plugins/dam/components/search-assets-screen/index.js:19]: @rsumilang can you call it `createRowClickHandler.js` and the data attribute to `data-ignore-row-click`

File: src/plugins/dam/components/search-assets-screen/index.js [hunk 4]
```diff
@@ -96,11 +99,12 @@ function SearchAssetsScreen(props) {
                 const canUpdate = policy.isGranted(`${schema.getQName()}:update`);
                 const seq = node.get('gallery_seq');
                 const transcodingStatus = node.get('transcoding_status');
+                const handleNodeClick = nodeClickNavigator(navigate, node);
```
Comment (gdbrown) [inline src/plugins/dam/components/search-assets-screen/index.js:102]: @rsumilang call it `const handleRowClick = createRowClickHandler(navigate, node);`

File: src/plugins/dam/components/search-assets-screen/index.js [hunk 5]
```diff
@@ -96,11 +99,12 @@ function SearchAssetsScreen(props) {
...
 
                 return (
-                  <tr key={`${node.get('_id')}`} className={`status-${node.get('status')}`}>
-                    <td><Input type="checkbox" onChange={() => batch.toggle(node)} checked={batch.has(node)} /></td>
-                    <td className="text-center"><AssetIcon id={node.get('_id')} /></td>
+                  <tr key={`${node.get('_id')}`} className={`status-${node.get('status')}`} style={{cursor: 'pointer'}} onClick={handleNodeClick}>
```
Comment (gdbrown) [inline src/plugins/dam/components/search-assets-screen/index.js:105]: @rsumilang you won't need this style, i tested a fix on sloth branch and it seems to work in chrome.
```
<tr key={`${node.get('_id')}`} className={`cursor-pointer status-${node.get('status')}`}>
```

File: src/plugins/dam/components/search-assets-screen/index.js [hunk 6]
```diff
@@ -96,11 +99,12 @@ function SearchAssetsScreen(props) {
...
+                    <td data-ignore-node-click><Input type="checkbox" onChange={() => batch.toggle(node)} checked={batch.has(node)} /></td>
```
Comment (gdbrown) [inline src/plugins/dam/components/search-assets-screen/index.js:106]: @rsumilang use `data-ignore-row-click`

**Review by gdbrown — CHANGES_REQUESTED** [review]

File: src/plugins/dam/components/search-assets-screen/index.js [hunk 7]
```diff
@@ -96,11 +99,12 @@ function SearchAssetsScreen(props) {
                 const canUpdate = policy.isGranted(`${schema.getQName()}:update`);
                 const seq = node.get('gallery_seq');
                 const transcodingStatus = node.get('transcoding_status');
+                const handleRowClick = createRowClickHandler(navigate, node);
 
                 return (
-                  <tr key={`${node.get('_id')}`} className={`status-${node.get('status')}`}>
-                    <td><Input type="checkbox" onChange={() => batch.toggle(node)} checked={batch.has(node)} /></td>
-                    <td className="text-center"><AssetIcon id={node.get('_id')} /></td>
+                  <tr key={`${node.get('_id')}`} className={`status-${node.get('status')}`} onClick={handleRowClick}>
```
Comment (gdbrown) [inline src/plugins/dam/components/search-assets-screen/index.js:105]: @rsumilang can you add that css class as well?  maximum spice

**Pull request merged**